app/models/category.rb

```diff
@@ -1,4 +1,4 @@
 class Category < ApplicationRecord
     has_and_belongs_to_many :books
-    validates :name, unique:  true
+    validates :name, uniqueness:  true, allow_blank: false
```

Comment:
what do you mean by `allow_blank: false` ? I think you should replace it by `presence: true`

Reply:
- `allow_blank: false` means no empty string

- no category is not required

Reply:
have you tested both? because I think presence will also not allow empty string.
Resolving because looking into this is a minor and require diving deep in the Rails source code.